cli/src/api/mod.rs

```diff
@@ -296,9 +304,23 @@ impl PhylumApi {
             group_name,
         };
         log::debug!("==> Sending package submission: {:?}", req);
-        let resp: SubmitPackageResponse =
-            self.post(endpoints::post_submit_package(&self.config.connection.uri)?, req).await?;
-        Ok(resp.job_id)
+
+        let uri = endpoints::post_submit_package(&self.config.connection.uri)?;
+        let response: Result<SubmitPackageResponse> = self.post(uri, req).await;
+
+        response
+            .map_err(|err| match err {
+                PhylumApiError::Response(ResponseError { code: StatusCode::NOT_FOUND, .. }) => {
+                    AnyhowError::new(err)
+                        .context(format!(
+                            "Project with ID {project} does not exist. This usually means Phylum \
+                             is still linked to a project that has been deleted."
+                        ))
+                        .into()
+                },
```

Comment:
This may be obsoleted by phylum-dev/api#478

Reply:
Looks like it would be, yeah. Seems like the requests were submitted to API and CLI simultaneously somewhat.

I'm fine with merging this and then just ripping it out again, but I'd personally prefer we just scrap this PR and move on without it, so we don't have useless code churn.

I'd still like to move the other checks to the API since I think that's a better place for them.